tests/integ/kms_utils.py

```diff
@@ -42,22 +48,72 @@ def _get_kms_key_arn(kms_client, alias):
         return None
 
 
-def _create_kms_key(kms_client, account_id):
+def _get_kms_key_id(kms_client, alias):
+    try:
+        response = kms_client.describe_key(KeyId='alias/' + alias)
+        return response['KeyMetadata']['KeyId']
+    except kms_client.exceptions.NotFoundException:
+        return None
+
+
+def _create_kms_key(kms_client,
+                    account_id,
+                    role_arn=None,
```

Comment:
why accept a role name and a role arn?

Reply:
There are three principals that need to be created: the account root, the role that is running the tests, and the role that will be used by SageMaker. Example:

```json
            "Principal": {
                "AWS": [
                    "arn:aws:iam::142577830533:role/SageMakerRole",
                    "arn:aws:sts::142577830533:assumed-role/DevBuildStackPDX-PullRequestBuildRole4B40B95C-G4AE4UBWNU2K/AWSCodeBuild-fe90a41f-9006-4ccc-918c-437c0f5b26d2",
                    "arn:aws:iam::142577830533:root"
                ]
            },
```